frontend/src/metabase/lib/urls/utils.ts

```diff
@@ -28,3 +30,24 @@ export function getEncodedUrlSearchParams(query: Record<string, unknown>) {
     })
     .join("&");
 }
+
+export function getSubpathSafeUrl(url?: string) {
+  if (url === undefined) {
+    return;
+  }
+
+  return api.basename + url;
+}
+
+/**
+ * Metabase can be deployed on a subpath!
+ * If you're opening internal links in a new tab, make sure you're using subpath-safe URLs.
+ * @see {@link getSubpathSafeUrl}
+ */
+export const openInNewTab = (url?: string) => {
+  if (!url) {
+    return;
+  }
+
+  window.open(url, "_blank");
+};
```

Comment:
just curious, when Metabase is deployed on a subpath, do we still have `<base href="/" />` in the HTML?

Reply:
`src/metabase/server/routes/index.clj:23` seems to set the base href correctly based on the `site-url` setting. I think `window.open` respects the base tag.

Reply:
Ah, I think you already discussed this with Alex P. Never mind.

Reply:
So we're not using it directly anymore.
Do you think JSDoc comment is enough?

Alternatively, we can introduce a lint rule, but we're opening external links with `window.open` in a few cases as well. This warning is only for internal links.